copilot/fastapi/tests/test_kubernetes_api.py

```
@@ -0,0 +1,48 @@
+
+def test_kubernetes_proxy_failure(test_client):
+    """Test the Kubernetes proxy endpoint when the API is unavailable"""
+    # Mock httpx to raise an exception
+    with patch("httpx.AsyncClient") as mock_client:
+        mock_client_instance = MagicMock()
+        mock_client_instance.__aenter__.return_value.request.side_effect = httpx.HTTPError("Connection error")
+        mock_client.return_value = mock_client_instance
+        
+        # Mock the token file
+        with patch("builtins.open", MagicMock()):
+            with patch("fastapi.kubernetes_api.open", MagicMock()):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)